go/test/endtoend/vtgate/transaction/trxn_test.go

```diff
@@ -174,3 +174,36 @@ func TestTransactionIsolation(t *testing.T) {
 	utils.AssertMatches(t, conn1, "select id, msg from test where id = 1", `[[INT64(1) VARCHAR("foo")]]`)
 	utils.Exec(t, conn1, "rollback")
 }
+
+func TestTransactionCharacteristics(t *testing.T) {
+	defer cluster.PanicHandler(t)
```

Comment:
this test is leaving lingering state. I really like the pattern that we`ve started using for end to end tests:

```go
	mcmp, closer := start(t)
	defer closer()
```

Reply:
This does not need `mysql` connection for that reason, this test package does not use that pattern

Reply:
The snippet:

```go
mcmp, closer := start(t)
defer closer()
```

Does not necessarily mean that you're using vitess + standalone mysql to run your tests. I think what @systay is referring too is a common function for all the tests in the package that returns the MySQL connection and a clean up function. That common function would remove the following snippet from all the test functions of the package:

```go
	conn, err := mysql.Connect(ctx, &vtParams)
	require.NoError(t, err)
	defer conn.Close()
```

Reply:
☝️ listen to this man